ziti-agent/cmd/webhook/pods.go

```diff
@@ -35,578 +37,657 @@ const (
 	labelAppInstance  = "app.kubernetes.io/instance"
 	labelAppComponent = "app.kubernetes.io/component"
 
-	warnEmptyJWT        = "pod created without ziti enrollment token"
-	warnIdentityNameExtract = "ziti identity name not found in pod annotations"
-	warnIdentityListFailed = "ziti identity list operation failed in ziti edge"
+	// warnEmptyJWT            = "pod created without ziti enrollment token"
+	// warnIdentityNameExtract = "ziti identity name not found in pod annotations"
+	// warnIdentityListFailed  = "ziti identity list operation failed in ziti edge"
+
+	zitiTypeRouter zitiType = "router"
+	zitiTypeTunnel zitiType = "tunnel"
 )
 
-type JsonPatchEntry struct {
-	OP    string          `json:"op"`
-	Path  string          `json:"path"`
-	Value json.RawMessage `json:"value,omitempty"`
+type zitiType string
+
+type clusterClient struct {
+	client *kubernetes.Clientset
+}
+
+type clusterClientIntf interface {
+	getClusterService(ctx context.Context, namespace string, name string, opts metav1.GetOptions) (*corev1.Service, error)
+	findNamespaceByOption(ctx context.Context, name string, opts metav1.ListOptions) (bool, error)
+}
+
+type zitiClient struct {
+	client *rest_management_api_client.ZitiEdgeManagement
+}
+
+type zitiClientIntf interface {
+	createIdentity(ctx context.Context, name string, key string, pod *corev1.Pod) (string, error)
+	deleteIdentity(ctx context.Context, id string) error
+	deleteZitiRouter(ctx context.Context, name string) error
+	getIdentityToken(ctx context.Context, name string, id string) (string, error)
+	getZitiRouterToken(ctx context.Context, name string) (string, error)
+	findIdentityId(ctx context.Context, name string) (string, error)
+	patchIdentityRoleAttributes(ctx context.Context, id string, key string, newPod *corev1.Pod, oldPod *corev1.Pod) error
+	updateZitiRouter(ctx context.Context, name string, options *rest_model_edge.EdgeRouterCreate) (*edge_router.CreateEdgeRouterCreated, error)
+}
+
+type zitiConfig struct {
```

Comment:
Same struct for tunnelers and routers means tunnelers have a RouterConfig. Do tunnelers need a RouterConfig?

Reply:
no, It is only for routers, but I just initialized it to null for best practice.